### ISSUE TITLE
Can't activate the OpenDRIVE plugin

### ISSUE THREAD

**Issue by KarimGhazalEXT** [brifsttar/OpenDRIVE#15]
I am using Windows 10, Unreal engine 4.26.2
I want to edit a CARLA map using unreal editor but when I add the repo files in the U4.26 plugins and check the enable box I am asked to restart the Engine and the this message pops up and the Engine doesn't start. 
![image](https://github.com/brifsttar/OpenDRIVE/assets/144326423/1d7c7b18-3da6-4c91-b51d-94fb678a7912)

**Comment by brifsttar** [comment]
Hi,

Did you add the plugin to the engine or the project directory? In any case, the plugin needs to be built from source, as no binaries are distributed.

You might also want to check #10 since there are some compatibility issues with CARLA.

**Comment by KarimGhazalEXT** [comment]
I added the plugin(the files extracted from the repo's zip) to the Engine's plugin directory.
How can I build it from source?

**Comment by brifsttar** [comment]
If you're using a pre-built version of the engine, I actually don't know if you can build a specific plugin from it. The easiest way would be to instead put the plugin in your project's. To build your project's code (and its plugins), you can generate the `.sln` by right clicking your `.uproject` and selecting `Generate Visual Studio project files`. This probably requires a C++ project though, and some experience with Visual Studio.

**Comment by kwcnldnsldskcls** [comment]
Hello, I have the same problem with UE 5.3.1. Could you provide a better solution please? I've tried everything with UE and VS but i can't get it to work... Thank you!

**Comment by brifsttar** [comment]
Hi @kwcnldnsldskcls ,

I'll try to summarize the complete installation steps. They require a C++ projects as starting point.

1. Clone the repository in your project's `Plugins/` folder (see https://github.com/brifsttar/OpenDRIVE#installation)
2. Right click your project's `.uproject` file and select `Generate Visual Studio project files`
3. Open the generated `.sln` file
4. In Visual Studio's Solution Explorer, right click your project and select `Build`
5. Launch your project

If you encounter any issue following these steps, please provide as much information as possible about the problem.

**Comment by kwcnldnsldskcls** [comment]
Thanks for the fats response! I've tried again and came some steps further. I've added the esmini thats linked in your project to the OpenDrive Third Party RoadManager Modules and now some errors went by but still a lot of files missing (Include Errors). Some I Could fix but it keeps going on and also sends me some Class Errors. Pretty complicated... Have you any ideas how to easliy solve this? Thank you!

**Comment by brifsttar** [comment]
Could you share those errors? That would be of great help to understand the issue.

**Comment by kwcnldnsldskcls** [comment]
Hey! After some more research I found out that the OpenDriveComponent.GENERATED.h is not in any directory. This file is included in the OpenDriveComponent.h. I guess thats the missing one. Do you have the file?

**Comment by brifsttar** [comment]
This file should be generated automatically when you generate the solution files.

**Comment by kwcnldnsldskcls** [comment]
Getting closer to the solution I guess... I made a complete fresh Project and now only these errors occur
![Unbenannt](https://github.com/brifsttar/OpenDRIVE/assets/126483307/c3b524cb-89a0-471c-bc0a-2ef7500f670d)

**Comment by kwcnldnsldskcls** [comment]
Thanks mate! That worked!

**Comment by brifsttar** [comment]
That's weird... I'm not sure about 4.27, but it should definitely work with 5.3.1 (I just tested to be sure).

Can you open/compile the asset `/OpenDRIVE/EUBP_Actor`?

Do you have the "Editor Scripting Utilities" plugin enabled?

![image](https://github.com/brifsttar/OpenDRIVE/assets/53508707/8e99dbee-e668-4ea0-b6b6-4bf922070886)

**Comment by kwcnldnsldskcls** [comment]
Yep the Plugin is enabled and VS compiled everything without an error…

**Comment by brifsttar** [comment]
Can you check in the Content Browser, under the Plugin subsection (you need to enable `Show Plugin Content` in the browser settings), then under `OpenDRIVE Content`, you should find `EUBP_Actor`. Open the asset, try to compile it, and then see if the actions are available. This BP actually contains those, so maybe it's not registering properly.

**Comment by kwcnldnsldskcls** [comment]
That one isn’t there but in the Project Plugin Folder the uasset is shown. 
But it worked for some minutes weird… I can remember I tried creating a new EUBP_Actor, Editor and Object in my content compiled it and it worked somehow (maybe I did something else also) but then it was gone again and with this steps I wasn’t able to reproduce that the action is showing again

**Comment by brifsttar** [comment]
> That one isn’t there but in the Project Plugin Folder the uasset is shown.
You mean that in the Content Browser, you can't find the `EUBP_Actor`, but in the Windows Explorer, you can see the `EUBP_Actor.uasset`?

If so, could you tell me the size of this file? Since the repository uses Git LFS to track `.uasset`, it's possible that it wasn't downloaded properly.

**Comment by kwcnldnsldskcls** [comment]
Exactly! The file size is 131 Bytes

**Comment by brifsttar** [comment]
Do you have Git LFS installed? You can just run `git lfs` to check.

If it's installed, you might want to run a `git reset --hard` to reset the repository to its default, it should (maybe?) redownload assets.

**Comment by kwcnldnsldskcls** [comment]
I don’t have lfs installed I will do a fresh project , download everything again and check if it works. I mean it worked so it should be no problem but weird that the actions were shown for some time… I’ll let you know about the results… 🙏🏼

**Comment by kwcnldnsldskcls** [comment]
So it’s still the same… Building the plugin now works properly no errors or stuff but I don’t know why I can’t access the Actions I tried creating new actions for the landscape to maybe trigger the “hidden menue” 

Could you provide a step by step what I should click to access the scripted action for example select landscape in outliner -> go to landscape-> sculpt or something?

**Comment by brifsttar** [comment]
Before going further: do you now see the EUBP_Actor in the Content Browser? If so, can you open it and compile it? If not, is it visible in the Windows Explorer? If so, what is its file size?

**Comment by kwcnldnsldskcls** [comment]
Still same in WE it’s shown in CE it’s not visible. 
131Bytes

I can create a new Editor Utility Actor or Object or Asset and compile it.

**Comment by brifsttar** [comment]
It seems something is going wrong with your clone, because LFS objects aren't being downloaded. Could you maybe rerun a clone and share the log? Or run a `git status`?

I've enabled LFS exports on archive for this project, so if you simply download a zip of the project, it should include the correct files.

**Comment by kwcnldnsldskcls** [comment]
After an enternity it finally works 100%! Thank you so much!

**Comment by hasanzfdn** [comment]
> Hi @kwcnldnsldskcls ,
> 
> I'll try to summarize the complete installation steps. They require a C++ projects as starting point.
> 
> 1. Clone the repository in your project's `Plugins/` folder (see https://github.com/brifsttar/OpenDRIVE#installation)
> 2. Right click your project's `.uproject` file and select `Generate Visual Studio project files`
> 3. Open the generated `.sln` file
> 4. In Visual Studio's Solution Explorer, right click your project and select `Build`
> 5. Launch your project
> 
> If you encounter any issue following these steps, please provide as much information as possible about the problem.

![image](https://github.com/brifsttar/OpenDRIVE/assets/69383294/02702a09-9f9c-43fc-b61f-a5f3024f9d63)
I had same error at first after that while trying to build I had this error . Could you help me about that?

**Comment by brifsttar** [comment]
Hi,

Could you share the complete output (Output tab next to "Error List"). The error displayed is the very generic Unreal message, the actual error will only be in the Output.

**Comment by hasanzfdn** [comment]
![image](https://github.com/brifsttar/OpenDRIVE/assets/69383294/1d489d59-6aa4-4318-a85b-8f5c1405f36e)

Sorry for late reply. it says no existing makefile

**Comment by brifsttar** [comment]
Can you maybe redo step 2 mentioned above (*Right click your project's .uproject file and select Generate Visual Studio project files*) and check that this is succesful?

Also, could you maybe run a `git status` in the OpenDRIVE plugin folder to check that everything was properly cloned?

**Comment by hasanzfdn** [comment]
1) used this link to get packages (git clone --recurse-submodules https://github.com/brifsttar/OpenDRIVE.git) and git status showed me already up to date with origin/master.
2) created blank simulation project with c++
3) added OpenDRIVE plugin and restart, the same error again
4) Right click your project's .uproject file and select Generate Visual Studio project files
5) In Visual Studio's Solution Explorer, right click your project and select Build
and error be like
![image](https://github.com/brifsttar/OpenDRIVE/assets/69383294/6317bcc2-f2ec-4e8c-82aa-8b820a5c03f4)
